### PR TITLE
search done

### PR DESCRIPTION
**Pull request by Kevin-Zh-CS**
You can input the disease, we can return the Department information that you can hang up, and it is the Department that we have defined in the database

### CODE REVIEW

**Comment by AsterNighT** [comment]
pls address the conflict first!

File: pkg/process/model.go [hunk 1]
```diff
@@ -94,6 +94,13 @@ const (
 	CannotAssignDoctor       ProcessError = "无法为该挂号分配医生"
 	AccountNotFound          ProcessError = "找不到用户"
 	RegistrationUpdateFailed ProcessError = "挂号状态更新失败"
+	SearchFailed		     ProcessError = "查询失败"
+)
+
+type SearchKey string
+const (
+	KevinKey	SearchKey = "AO6VqIdnT4z+h5mO9Ig3QmvGMwTgsJeZ/pxz7Pw"
```
Comment (AsterNighT) [inline pkg/process/model.go]: You'd better not put the key directly into code base. Check account module for an example.

**Comment by AsterNighT** [comment]
Pls fix the checks

File: pkg/process/handler.go [hunk 2]
```diff
@@ -1,17 +1,61 @@
...
 	"gorm.io/gorm"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"strings"
 )
 
-type ProcessHandler struct{}
+type Handler struct{}
```
Comment (ziyue-pan) [inline pkg/process/handler.go]: Better be `ProcessHandler`

**Review by ziyue-pan — CHANGES_REQUESTED** [review]

File: pkg/process/model.go (hunk 1 above)
Comment (Kevin-Zh-CS) [reply pkg/process/model.go]: not private key, not private info, do not need to write into conf file.

File: pkg/router/router.go [hunk 3]
```diff
@@ -101,6 +101,7 @@ func RegisterRouters(app *echo.Echo) error {
 			router.POST("/milestones", h.CreateMileStoneByDoctor)
 			router.PUT("/milestone/:mileStoneID", h.UpdateMileStoneByDoctor)
 			router.DELETE("/milestone/:mileStoneID", h.DeleteMileStoneByDoctor)
+			router.GET("/search/:KeyWord", h.Search)
```
Comment (AsterNighT) [inline pkg/router/router.go]:
```suggestion
			router.GET("/search/:keyWord", h.Search)
```

File: pkg/process/handler.go [hunk 4]
```diff
@@ -604,3 +650,15 @@ func (h *ProcessHandler) DeleteMileStoneByDoctor(c echo.Context) error {
 	db.Delete(&mileStone)
 	return c.JSON(http.StatusOK, api.Return("ok", nil))
 }
+
+func Transformation(response *http.Response) map[string]interface{} {
```
Comment (AsterNighT) [inline pkg/process/handler.go]: This procedure is more precisely called "parsing" or "deserialize"

**Review by AsterNighT — CHANGES_REQUESTED** [review]

**Review by AsterNighT — APPROVED** [review]

**Comment by AsterNighT** [comment]
@PAN-Ziyue If you have no more concerns, pls approve the PR.

**Review by ziyue-pan — APPROVED** [review]

**Pull request merged**